src/pybamm/expression_tree/averages.py

```diff
@@ -394,13 +406,35 @@ def size_average(
     else:
         if f_a_dist is None:
             geo = pybamm.geometric_parameters
+            if "negative" in symbol.domain[0]:
+                if "primary" in symbol.domain[0]:
+                    name = "R_n_prim"
+                elif "secondary" in symbol.domain[0]:
+                    name = "R_n_sec"
+                else:
+                    name = "R_n"
+            else:
+                if "primary" in symbol.domain[0]:
+                    name = "R_p_prim"
+                elif "secondary" in symbol.domain[0]:
+                    name = "R_p_sec"
+                else:
+                    name = "R_p"
             R = pybamm.SpatialVariable(
-                "R", domains=symbol.domains, coord_sys="cartesian"
+                name, domains=symbol.domains, coord_sys="cartesian"
             )
-            if ["negative particle size"] in symbol.domains.values():
+            if ["negative particle size"] in symbol.domains.values() or [
```

Comment:
FYI `geo.domain_params["negative"]` is equivalent to `geo.n`

Reply:
I'm not sure how to interpret this comment, as far as I can tell, I never use `geo.domain_params['negative']`. Is there something here you want changed?

Reply:
sorry I wasn’t clear. I meant that instead of doing an if statement by domain and getting eg `geo.n` you can do `geo.domain_params[domain]`. More of a comment then necessarily a suggestion / required change.

Reply:
looks like I left this comment in the wrong place too lol

Reply:
sounds good, thanks